servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/RoundRobinSelector.java

```diff
@@ -75,7 +73,7 @@ protected Single<C> selectConnection0(
             }
         }
         // We were unable to find a suitable host.
-        return noActiveHostsFailure(usedHosts);
+        return noActiveHostsFailure(hosts());
```

Comment:
Will it make sense to capture `hosts()` in a local variable to make sure we always look at exact same `hosts` view for a single selection?